src/hott/types/2_adj/hty_rec.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Authors: Daniel Carranza, Jonathon Chang, Ryan Sandford
+Under the supervision of Chris Kapulkin
+
+Contains theorems for performing induction on homotopies which
+  is used in adj.lean and two_adj.lean to prove 
+  full-adjoint equivalences are not mere propositions
+
+Last updated: 2020-06-12
+-/
+
+import hott.init hott.types.equiv
+open hott
+
+hott_theory
+universes u v
+
+namespace homotopy
+
+  variable {A : Type u}
+  variable {B : A → Type v}
+
+  @[hott] protected def rec' (f : Π(x : A), B x) (P : Π(g : Π(x : A), B x), (f = g) → Type _) (c : P f rfl)
+    : Π(g : Π(x : A), B x) (p : f = g), P g p :=
+  λg p, by hinduction p; exact c
+
+  -- Based homotopy induction
+  @[hott, induction] def rec {f : Π(x : A), B x} (P : Π(g : Π(x : A), B x), (f ~ g) → Type _) (c : P f (λx, refl (f x)))
+    : Π(g : Π(x : A), B x) (H : f ~ g), P g H :=
```

Comment:
I think this already exists: https://github.com/gebner/hott3/blob/436a4f5e49f51ca5773b34df046bbf67e2584f70/src/hott/init/funext.lean#L291-L292